### ISSUE TITLE
pysat fails to import due to pandas AttributeError

### ISSUE THREAD

**Issue by asreimer** [pysat/pysat#730]
**Describe the bug**
Simply attempting `import pysat` results in the following `panda`s error:
```
[asreimer@localhost ~]$ python
Python 3.7.9 (default, Sep 22 2020, 09:19:36) 
[GCC 10.2.1 20200826 (Red Hat 10.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pysat
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/asreimer/test/lib/python3.7/site-packages/pysat/__init__.py", line 94, in <module>
    from . import instruments
  File "/home/asreimer/test/lib/python3.7/site-packages/pysat/instruments/__init__.py", line 20, in <module>
    from . import *
  File "/home/asreimer/test/lib/python3.7/site-packages/pysat/instruments/pysat_testing.py", line 219, in <module>
    meta = pysat.Meta()
  File "/home/asreimer/test/lib/python3.7/site-packages/pysat/_meta.py", line 228, in __init__
    self._fill_label])
  File "/home/asreimer/test/lib64/python3.7/site-packages/pandas/core/frame.py", line 392, in __init__
    mgr = init_dict(data, index, columns, dtype=dtype)
  File "/home/asreimer/test/lib64/python3.7/site-packages/pandas/core/internals/construction.py", line 196, in init_dict
    nan_dtype)
  File "/home/asreimer/test/lib64/python3.7/site-packages/pandas/core/dtypes/cast.py", line 1175, in construct_1d_arraylike_from_scalar
    dtype = dtype.dtype
AttributeError: type object 'object' has no attribute 'dtype'
```

**To Reproduce**
1. Create a python 3.7 virtual environment: `virtualenv -p python3.7 test`
2. Source it: `source test/bin/activate`
3. `pip install pysat`
4. Open a python shell and import pysat: `python -c 'import pysat'`

**Expected behavior**
`pysat` should import without throwing an exception.

**Desktop:**
 - OS: Fedora 33
 - Python 3.7.9

I have tried a few different versions of pandas (0.23.1, 0.24.0, etc) and I still get the same error. Related question, why does pysat restrict to such an old version of pandas (<0.25,>=0.23)?

**Comment by asreimer** [comment]
In case it's useful, here's the output of `pip freeze`:

```
beautifulsoup4==4.9.3
cached-property==1.5.2
certifi==2020.12.5
cftime==1.4.1
chardet==4.0.0
cycler==0.10.0
h5py==3.2.1
idna==2.10
kiwisolver==1.3.1
lxml==4.6.2
madrigalWeb==3.2
matplotlib==3.3.4
mpmath==1.2.1
netCDF4==1.5.6
numpy==1.20.1
pandas==0.24.2
Pillow==8.1.2
PyForecastTools==1.1.1
pyparsing==2.4.7
pysat==2.2.2
pysatCDF @ git+https://github.com/asreimer/pysatCDF.git@8ab4b5942f63bb68c1329b728cb429aa24f0d0cb
python-dateutil==2.8.1
pytz==2021.1
requests==2.25.1
scipy==1.6.1
six==1.15.0
soupsieve==2.2
urllib3==1.26.3
xarray==0.14.1
```

**Comment by asreimer** [comment]
I just tested with `pip install git+https://github.com/pysat/pysat.git@develop-3` and noticed that this doesn't result in the same `AttributeError`, but instead works as expected (yay!). Looks like [this commit](https://github.com/pysat/pysat/commit/b9c54d6b73eccb43d21791a39fb22047e18e34f2) may have accidentally fixed this problem?

**Comment by aburrell** [comment]
I love it when a new class I introduce solves more problems than the ones that were intended 😄 I think the rest of this issue may be important for the 2.3.0 deprecation release.

@asreimer if you're using pysat right now I'd recommend using the Release Candidate branch: https://github.com/pysat/pysat/pull/676.  We'd love to have feedback before the merge!

**Comment by rstoneback** [comment]
Thanks @asreimer for the report. pysat v2.x has an old pandas requirement since we have some code that uses the now removed Panel. pysat v3 removes the pandas restriction since the code that used Panel is now over in pysatSeasons.

**Comment by jklenzing** [comment]
Running some tests on this, it seems that pandas 0.24.2 is not compatible with numpy 1.20.1.  Trying to import pysat with numpy 1.20.1 produces the same error as above, but 1.19.2 is fine.

@rstoneback, @aburrell, this issue was hidden from us because numpy 1.20 is not in the conda environment, so the unit tests  use numpy 1.19.2 rather than the latest.  I'll dig through the docs in search of the change, but we should probably put a version cap on the pysat 2.3 branch for numpy.

**Comment by jklenzing** [comment]
@asreimer, we've just pushed a version cap on numpy to the `develop` (pysat 2.x) branch.  This should fix the above issue for a clean install of 2.3 when it's released.  pysat 3.0 is compliant with newer versions of pandas, so this shouldn't affect things there (as your above test indicates).  Let us know if there's still a problem here we've missed.
